diff --git a/xos/gui/Dockerfile b/xos/gui/Dockerfile
index 4b5ec97..60f7a50 100644
--- a/xos/gui/Dockerfile
+++ b/xos/gui/Dockerfile
@@ -9,7 +9,7 @@ COPY ${CODE_SOURCE}/package.json ${CODE_DEST}/package.json
 COPY ${CODE_SOURCE}/typings.json ${CODE_DEST}/typings.json
 
 WORKDIR ${CODE_DEST}
-RUN npm install
+RUN npm dedupe && npm install
 RUN npm run typings
 
 COPY ${CODE_SOURCE}/conf ${CODE_DEST}/conf
