diff --git a/Dockerfile b/Dockerfile
index 9f39abf..db6b24e 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -20,7 +20,7 @@ COPY ${CODE_SOURCE}/gulp_tasks/ ${CODE_DEST}/gulp_tasks/
 COPY ${CODE_SOURCE}/src/ ${CODE_DEST}/src/
 
 WORKDIR ${CODE_DEST}
-RUN npm install \
+RUN npm dedupe && npm install \
  && npm run typings \
  && npm run build
 
diff --git a/Dockerfile.xos-gui-builder b/Dockerfile.xos-gui-builder
index 32508ca..dad3325 100644
--- a/Dockerfile.xos-gui-builder
+++ b/Dockerfile.xos-gui-builder
@@ -16,19 +16,35 @@
 
 # To build use: docker build -t xosproject/xos-gui-builder:candidate .
 
-FROM node:7.9.0
+FROM node:8.11.1-stretch
 
 # Set environment vars
 ENV CODE_SOURCE .
 ENV CODE_DEST /var/www
+ENV QT_QPA_PLATFORM minimal
 
 RUN mkdir -p ${CODE_DEST}
 
 COPY ${CODE_SOURCE}/package.json ${CODE_SOURCE}/typings.json ${CODE_DEST}/
 
+RUN apt-get update \
+    && apt-get remove -y libssl-dev \
+    && apt-get install -y build-essential g++ flex bison \
+    gperf ruby perl libsqlite3-dev libfontconfig1-dev libicu-dev \
+    libfreetype6 libpng-dev libjpeg-dev libx11-dev libxext-dev zip libssl1.0-dev \
+    && git clone git://github.com/ariya/phantomjs.git \
+    && cd phantomjs \
+    && git checkout 2.1 \
+    && git submodule init \
+    && git submodule update \
+    && cd /phantomjs \
+    && python ./build.py --qt-config "-I /usr/include/openssl-1.0/ -L/usr/lib/openssl-1.0/" \
+    && apt-get update && apt-get install -y phantomjs
+
 WORKDIR ${CODE_DEST}
 RUN npm install \
- && npm run typings
+ && npm run typings \
+ && apt-get remove -y libssl1.0-dev && apt-get install -y libssl-dev
 
 ARG org_label_schema_schema_version=1.0
 ARG org_label_schema_name=xos-gui-builder
